desktop/ui/src/main/java/org/datacleaner/user/DataCleanerHome.java

```diff
@@ -97,17 +119,37 @@ private static FileObject findDataCleanerHome() throws FileSystemException {
             logger.info("Running in WebStart mode. Attempting to build DATACLEANER_HOME in user.home: {} -> {}", path,
                     candidate);
         } else {
-            // in normal mode, the default folder will be in the working
-            // directory
-            candidate = manager.resolveFile(".");
+            // in normal mode it is trying to use specified directory first,
+            // working directory as a fallback or user home directory as a
+            // second fallback.
+            logger.info("Running in standard mode.");
             if (isWriteable(candidate)) {
-                logger.info("Running in standard mode. Attempting to build DATACLEANER_HOME in '.' -> {}", candidate);
+                logger.info("Attempting to build DATACLEANER_HOME in {}", candidate);
             } else {
-                final String path = getUserHomeCandidatePath();
-                candidate = manager.resolveFile(path);
-                logger.info(
-                        "Application directory is not writeable. Attempting to build DATACLEANER_HOME in user.home: {} -> {}",
-                        path, candidate);
+                // Workaround: isWritable is not reliable for a non-existent
+                // directory. Just create it and check again.
+                logger.debug(
+                        "DATACLEANER_HOME ({}) appears to be not writable. Applying a workaround: creating this directory and checking again.",
+                        candidate);
+                if ((candidate != null) && (!candidate.exists())) {
+                    candidate.createFolder();
+                }
+                if (isWriteable(candidate)) {
+                    logger.info("Attempting to build DATACLEANER_HOME in {}", candidate);
```

Comment:
This log statement is still identical to line 127, because it is the same scenario - we just need to check it twice because of the workaround. Adding a log statement for the workaround presence with DEBUG level too.